scripts/release.py

```diff
@@ -0,0 +1,276 @@
+import re
+import json
+import atexit
+import collections
+import argparse
+from pathlib import Path
+import sys
+from packaging import version
+
+import pygit2
+import github
+
+
+class GithubUserResolver:
+    def __init__(self, email2commit, token):
+        self.__cache_file = Path(__file__).parent / "gh-users-cache.json"
+        self.__cache = (
+            json.loads(self.__cache_file.read_text())
+            if self.__cache_file.exists()
+            else {}
+        )
+        # filter unknown users hoping we'd be able to find them this time
+        self.__cache = {key: value for key, value in self.__cache.items() if value}
```

Comment:
qq: why are we using `__` as the prefix for the class variables here? From what I know, this is used to prevent name conflicts. Is it necessary here?

Reply:
Right, I meant instance attributes. My understanding was that private instance attributes are usually done with just one `_` instead of two?

Reply:
that's usually for "protected", and it's only a convention (not enforced, only linting tools complain if you break that - interpreter does nothing to prevent you from accessing the attribute from outside)